pkg/cache/informer_cache.go

```diff
@@ -193,8 +193,8 @@ func indexByField(indexer Informer, field string, extractor client.IndexerFunc)
 		rawVals := extractor(obj)
 		var vals []string
 		if ns == "" {
-			// if we're not doubling the keys for the namespaced case, just re-use what was returned to us
-			vals = rawVals
+			// if we're not doubling the keys for the namespaced case, just create a new slice with same length
```

Comment:
Sorry I don't understand. If you write a function that returns a pointer, you are not supposed to keep a reference to that pointer or its underlying value.

Reply:
Can you clarify why you are doing that?

Reply:
The story is, I compute and put the indexer values into the object status `.status.indexerValues`, so the extractor will be
```
func(obj client.Object) []string {
    someObject, _ := obj.(*group.ObjectKind)
    return someObject.Status.IndexerValues
}
```
As we know, the object from client cache should be unchangeable.

Just think another scene,
```
values := []string{"a", "b", "c"}
do(values)
doAgain(values)
```
Do we expect `values` will be changed by `do`? Then `doAgain` will get unexpected values.

Though slice(extractor return) is a pointer type here, but it is used because it contains immutable values, so I don't think it should be changed when being used.

Reply:
Slice(extractor return) here is more like a container containing immutable values(indexer values).

Reply:
Of course, will do it ~